collectors/debugfs.plugin/sys_devices_virtual_powercap.c

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "debugfs_plugin.h"
+
+#define METRIC_ID "system.power_consumption"
+
+struct measurement_t {
+    unsigned long long energy_uj;
+    usec_t time_us;
+};
+
+struct zone_t {
+    char name[FILENAME_MAX + 1];
+    char path[FILENAME_MAX + 1];
+    unsigned long long max_energy_range_uj;
+    struct measurement_t measurement;
+};
+
+static int g_zones_count = 0;
+static struct zone_t *g_zones = NULL;
+
+static int get_measurement(const char *path, struct measurement_t *measurement)
+{
+    int result;
+    if (likely((result = read_single_number_file(path, &measurement->energy_uj)) == 0)) {
+        measurement->time_us = now_monotonic_high_precision_usec();
+    }
+    return result;
+}
+
+static double
+calculate_watts(unsigned long long max_energy_range_uj, struct measurement_t *before, struct measurement_t *after)
+{
+    unsigned long long energy_uj = 0;
+    usec_t delta_us = after->time_us - before->time_us;
+    if (unlikely(delta_us == 0)) {
+        return 0;
+    }
+
+    if (likely(after->energy_uj >= before->energy_uj)) {
+        energy_uj = after->energy_uj - before->energy_uj;
+    } else {
+        energy_uj = after->energy_uj + (max_energy_range_uj - before->energy_uj);
+    }
+    return (double)energy_uj / delta_us;
+}
+
+static struct zone_t *
+get_zone(const char *control_type, struct zone_t *parent, const char *dirname, struct zone_t **zones, size_t *count)
+{
+    char temp[FILENAME_MAX + 1];
+    snprintfz(temp, FILENAME_MAX, "%s/%s", dirname, "name");
+
+    char name[FILENAME_MAX + 1] = "";
+    if (read_file(temp, name, sizeof(name) - 1) != 0) {
+        return NULL;
+    }
+
+    char *trimmed = trim(name);
+    if (unlikely(trimmed == NULL || trimmed[0] == 0)) {
+        return NULL;
+    }
+
+    snprintfz(temp, FILENAME_MAX, "%s/%s", dirname, "max_energy_range_uj");
+    unsigned long long max_energy_range_uj = 0;
+    if (unlikely(read_single_number_file(temp, &max_energy_range_uj) != 0)) {
+        collector_error("Cannot read %s", temp);
+        return NULL;
+    }
+
+    snprintfz(temp, FILENAME_MAX, "%s/%s", dirname, "energy_uj");
+    struct measurement_t measurement;
+    if (unlikely(get_measurement(temp, &measurement) != 0)) {
+        collector_error("%s: Cannot read %s", trimmed, temp);
+        return NULL;
+    }
+
+    int parent_idx = -1;
+    if (parent != NULL) {
+        parent_idx = parent - *zones;
+    }
+
+    *zones = (struct zone_t *)reallocz(*zones, sizeof(struct zone_t) * (*count + 1));
+    if (unlikely(*zones == NULL)) {
```

Comment:
The functions that we use to allocate data does not need these kind of tests, because the function already test the result and if there is not memory it will close the software as you can see [here](https://github.com/netdata/netdata/blob/f48454dc6a5aad597390f6855d6e990e85c0e8fd/libnetdata/libnetdata.c#L470-L474), please,  can you remove this tests from your code?